src/indexer/merger.rs

```diff
@@ -519,7 +574,7 @@ impl IndexMerger {
         fast_field_serializer: &mut CompositeFastFieldSerializer,
         doc_id_mapping: &Option<Vec<(DocId, SegmentReaderWithOrdinal)>>,
         reader_and_field_accessors: &[(&SegmentReader, T)],
-    ) -> crate::Result<()> {
+    ) -> crate::Result<Vec<u64>> {
```

Comment:
Can we document in the rustdoc what the return value is? (If I understood correctly it is the index.)